Jenkinsfile

```
@@ -744,6 +744,9 @@ def getCommonModuleInTheChangeSet(String directory) {
   def exclude = "^(${directoryExclussion}|((?!\\/module\\/).)*\$|.*\\.asciidoc|.*\\.png)"
   dir("${env.BASE_DIR}") {
     module = getGitMatchingGroup(pattern: pattern, exclude: exclude)
+    if(!fileExists(module)) {
+      module = ''
+    }
```

the `getCommonModuleInTheChangeSet(String directory)` search for a pattern in a directory path, fi the pattern does not exist will return ''. I am not sure of the goal of check a folder with the name of the module exists in the current folder, probably will never exist.

cc @v1v

---

Is pattern checked for the current(=new) state or the old one? In this particular case we have removed the `openmetrics` module complete from x-pack but the tester still understands that is has changes and thus it kicks the tests for it which leads to test's failure later on since the module does not actually exists under x-pack.

---

A bit of context:

Unfortunately this is an edge case when moving files between x-pack and oss beats.

https://github.com/elastic/apm-pipeline-library/pull/535 didn't implement a git diff by excluding files that have been deleted/moved.

So somehow it tries to run the module in `x-pack` when that particular `module` was moved to the `oss`.

Therefore, the validation if the module folder exists could be a good workaround.

> I am not sure of the goal of check a folder with the name of the module exists in the current folder, probably will never exist.

if the module name exists in the `pwd` then it should work, unless `module` is a nested folder somewhere else, if so, it's required to consider that particular case.

---

I've just confirmed what Ivan said, and there is something missing in this logic, something like the below snippet might work, though I don't know if `module` is always a nester folder under `<beat_name_folder>` or `x-pack/<beat_name_folder>`

```suggestion
    if(!fileExists("${directory}/module/${module}")) {
      module = ''
    }
```